pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -0,0 +1,582 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enterprise
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	enterpriseApi "github.com/splunk/splunk-operator/pkg/apis/enterprise/v2"
+	splcommon "github.com/splunk/splunk-operator/pkg/splunk/common"
+	splctrl "github.com/splunk/splunk-operator/pkg/splunk/controller"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// handle to hold the pipeline
+var afwPipeline *AppInstallPipeline
+
+// createPipelineWorker creates a pipeline worker for an app package
+func createPipelineWorker(appDeployInfo *enterpriseApi.AppDeploymentInfo, appSrcName string, podName string, appFrameworkConfig *enterpriseApi.AppFrameworkSpec, client *splcommon.ControllerClient, cr splcommon.MetaObject, statefulSet *appsv1.StatefulSet) *PipelineWorker {
+	return &PipelineWorker{
+		appDeployInfo: appDeployInfo,
+		appSrcName:    appSrcName,
+		targetPodName: podName,
+		afwConfig:     appFrameworkConfig,
+		client:        client,
+		cr:            cr,
+		sts:           statefulSet,
+	}
+}
+
+// createAndAddAWorker used to add a worker to the pipeline on reconcile re-entry
+func (ppln *AppInstallPipeline) createAndAddPipelineWorker(phase enterpriseApi.AppPhaseType, appDeployInfo *enterpriseApi.AppDeploymentInfo, appSrcName string, podName string, appFrameworkConfig *enterpriseApi.AppFrameworkSpec, client splcommon.ControllerClient, cr splcommon.MetaObject, statefulSet *appsv1.StatefulSet) {
+	scopedLog := log.WithName("createAndAddPipelineWorker").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+	worker := createPipelineWorker(appDeployInfo, appSrcName, podName, appFrameworkConfig, &client, cr, statefulSet)
+
+	if worker != nil {
+		scopedLog.Info("Created new worker", "Pod name", worker.targetPodName, "App name", appDeployInfo.AppName, "digest", appDeployInfo.ObjectHash, "phase", appDeployInfo.PhaseInfo.Phase)
+		ppln.addWorkersToPipelinePhase(phase, worker)
+	}
+}
+
+// getApplicablePodNameForWorker gets the Pod name relevant for the CR under work
+func getApplicablePodNameForWorker(cr splcommon.MetaObject, podID int) string {
+	var podType string
+
+	switch cr.GetObjectKind().GroupVersionKind().Kind {
+	case "Standalone":
+		podType = "standalone"
+	case "LicenseMaster":
+		podType = "license-master"
+	case "SearchHeadCluster":
+		podType = "deployer"
+	case "IndexerCluster":
+		// error?
+	case "ClusterMaster":
+		podType = "cluster-master"
+	case "MonitoringConsole":
+		podType = "monitoring-console"
+	}
```

Comment:
For the greater good, can you please create a separate function, for e.g., `ToType()` (similar to `ToKind()` in pkg/splunk/enterprise/types.go`